proposals/nnnn-trailing-comma-lists.md

```diff
@@ -0,0 +1,294 @@
+# Allow trailing comma in comma-separated lists
+
+- Proposal: SE-NNNN
+- Author: Mateus Rodrigues
+- Review Manager: TBD
+- Status: Awaiting Review
+- Implementation: [swift](https://github.com/mateusrodriguesxyz/swift/commit/114b3b4ba1440d6f681b7b5f1086633b504eb2fb), [swift-syntax](https://github.com/mateusrodriguesxyz/swift-syntax/commit/17d3123f1868f450c878cb017403c4c686e8fc86)
+- Review: [pitch](https://forums.swift.org/t/pitch-allow-trailing-comma-in-tuples-arguments-and-if-guard-while-conditions/70170/48)
+
+## Introduction
+
+This proposal aims to allow the use of trailing commas, currently restricted to array and dictionary literals, in more comma-separated lists.
+
+## Motivation
+
+### Development Quality of Life Improvement
+
+A trailing comma is an optional comma after the last item in a list of elements:
+
+```swift
+let rank = [
+  "Player 1",
+  "Player 3",
+  "Player 2",
+]
+```
+
+Using trailing commas makes it easy to add, remove, reorder and comment in/out elements, without the need to add or delete the comma while doing any of these manipulations.
+
+Consider the following SwiftUI modifier:
+
+```swift
+func frame(
+    width: CGFloat? = nil,
+    height: CGFloat? = nil,
+    alignment: Alignment = .center
+) -> some View
+```
+
+`frame(width:)`, `frame(height:)`, `frame(width:alignment:)`, `frame(height:alignment:)`, `frame(width:height:)`, `frame(width:height:alignment:)` are all valid calls but you can't easily switch between `frame(width:)` and `frame(width:alignment:)` by commenting in/out `alignment` without adding/removing the trailing comma.
+
+```swift
+.frame(
+    width: 500,
+//    alignment: .leading
+) // ❌ Unexpected ',' separator
+```
+
+The introduction of [parameter packs](https://github.com/apple/swift-evolution/blob/main/proposals/0393-parameter-packs.md) allows more APIs that are comma-separated lists at call site and would benefit from trailing comma.
```

Comment:
Before this sentence, I would move the acknowledgment you have written below about SE-0084. You can write that a narrower proposal was reviewed and rejected, but since that time the language has evolved substantially that change the basis for rejection.

That would include not just evolution in prevailing code style, but also the point here that we have introduced parameter packs, but also the following point that we have introduced code generation tools.